### PR TITLE
fmt: upgrade 8.1.1 -> 10.2.1

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade 8.1.1 -> 10.2.1

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: [PR-7093](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=478241&view=results) ✅

### CODE REVIEW

File: SPECS/fmt/fmt.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "fmt-8.1.1.tar.gz": "3d794d3cf67633b34b2771eb9f073bde87e846e0d395d254df7b211ef1ec7346"
+  "fmt-10.1.0.tar.gz": "deb0a3ad2f5126658f2eefac7bf56a042488292de3d7a313526d667f3240ca0a"
  }
 }
```
Comment (christopherco) [inline SPECS/fmt/fmt.signatures.json:5]: nit - newline EOF
```suggestion
}

```

**Review by christopherco — APPROVED** [review]

File: SPECS/fmt/fmt.spec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Summary:        Small, safe and fast formatting library for C++
 Name:           fmt
-Version:        8.1.1
+Version:        10.1.0
```
Comment (christopherco) [inline SPECS/fmt/fmt.spec:3]: Curious if we can move to 10.1.1 which is the current release

Comment (mfrw) [reply SPECS/fmt/fmt.spec]: went to `10.2.1` which was more recent.

File: SPECS/fmt/fmt.signatures.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "fmt-8.1.1.tar.gz": "3d794d3cf67633b34b2771eb9f073bde87e846e0d395d254df7b211ef1ec7346"
+  "fmt-10.2.1.tar.gz": "1250e4cc58bf06ee631567523f48848dc4596133e163f02615c97f78bab6c811"
  }
 }
```
Comment (mfrw) [inline SPECS/fmt/fmt.signatures.json:5]:
```suggestion
}

```

**Pull request merged**